### ISSUE TITLE
[AppBarLayout] Snap behavior bug

### ISSUE THREAD

**Issue by Laimiux** [material-components/material-components-android#2001]
**Description:**
There is a bug with AppBarLayout in how it calculates `snapBottom` within `snapToChildIfNeeded`. The issue happens when the `AppBarLayout` has more than one child for example`CollapsingToolbarLayout` and another view below it such as `TabLayout`. After scrolling, the collapsed toolbar expands again automatically. 

The logical error is in the following code in `snapToChildIfNeeded`. 

```java
if (offsetChildIndex == abl.getChildCount() - 1) {
  // If this is the last child, we need to take the top inset into account
  snapBottom += abl.getTopInset();
}
```

The top inset is only used for the last child which in this case would be `TabLayout`. I think instead of the last child, the first child needs to take the top inset into account. So logic should be

```java
if (offsetChildIndex == 0) {
  snapBottom += abl.getTopInset();
}
```

**Expected behavior:** Screenshots and/or description of expected behavior

**Source code:** The code snippet which is causing this issue. Please consider attaching a minimal sample app that reproduces the issue.

**Android API version:** 30

**Material Library version:** 1.2.0

**Device:** Pixel 4a

**Comment by Khang-NT** [comment]
I have the same issue, the library has some bugs at method `snapToChildIfNeeded` because it doesn't take the`AppBarLayout#getTopInset` into account correctly, if I set `android:fitsSystemWindows="false"`, then it works.

**Comment by drchen** [comment]
It's fixed by @Khang-NT 's change

**Comment by Khang-NT** [comment]
> It's fixed by @Khang-NT 's change

Hi, will it be merged to branch 1.4?
